xlators/cluster/afr/src/afr-common.c

```diff
@@ -3459,7 +3459,11 @@ afr_lookup_selfheal_wrap(void *opaque)
     return 0;
 
 unwind:
-    AFR_STACK_UNWIND(lookup, frame, -1, EIO, NULL, NULL, NULL, NULL);
+    if (!local->xattr_rsp)
+        local->xattr_rsp = dict_new();
+    dict_copy(local->xattr_req, local->xattr_rsp);  // For gfid-heal-msg
```

Comment:
Copying request to response doesn't seem right. Need to check this a bit more.

Reply:
afr_selfheal_name() takes a single dict to both get (heal-op,child-name) and set (gfid-heal-msg) things. For this change, we could just copy gfid-heal-msg to response dict. Otherwise I think we need to  change afr_selfheal_name() signature to include 2 dicts.

Reply:
Setting response in xattr-req lead to bugs in the past for different xlators. I think setting the data directly in response would be better.

Reply:
I have used separate dicts.